# HG changeset patch
# User Rik <rik@octave.org>
# Date 1456884820 28800
#      Tue Mar 01 18:13:40 2016 -0800
# Node ID b9cb3a96849a8ce494912b7c115ab0a919ee7eea
# Parent  cd9d95d74403c62b311412c83d2bc44efb2e3409
build: Calculate ctags for m-files as well as C/C++/Fortran sources (bug #47317).

* scripts/module.mk: Populate magic variables TAG_DEPENDENCIES and TAG_FILES
with the list of m-files in the scripts directory.

diff --git a/scripts/module.mk b/scripts/module.mk
--- a/scripts/module.mk
+++ b/scripts/module.mk
@@ -153,21 +153,20 @@ check-missing-semicolon:
 	  for m in $(addprefix $(srcdir)/, $(FCN_FILES)) $(GEN_FCN_FILES); do \
 	    ! $(GREP) -q -E '^classdef' $$m || continue; \
 	    ! $(GREP) -q -E '^  *\<function\>' $$m || continue; \
 	    ! (echo $$m | $(GREP) -q __splinefit__.m) || continue; \
 	    echo "source ('$$m');"; \
 	  done ) | $(SHELL) run-octave --norc --silent --no-history
 .PHONY: check-missing-semicolon
 
-## Add rule to generate ctags.
-## Automake would normally generate such a rule, but only if there is a
-## xxx_SOURCES target
-ctags:
-	ctags $(addprefix $(srcdir)/, $(FCN_FILES)) $(GEN_FCN_FILES)
+## Include m-files in list of sources when building tag files.
+## Automake will not include these because there is no xxx_SOURCES target
+TAGS_DEPENDENCIES = $(addprefix $(srcdir)/, $(FCN_FILES)) $(GEN_FCN_FILES)
+TAGS_FILES = $(addprefix $(srcdir)/, $(FCN_FILES)) $(GEN_FCN_FILES)
 
 install-data-local: install-startup-files install-pkg-add
 
 uninstall-local: uninstall-startup-files uninstall-pkg-add
 
 install-pkg-add:
 	for f in $(PKG_ADD_FILES); do \
 	  if [ -n "`cat $$f`" ]; then \
